src/test/app/DepositAuth_test.cpp

```diff
@@ -724,14 +792,506 @@ struct DepositPreauth_test : public beast::unit_test::suite
         }
     }
 
+    void
+    testCredentials()
+    {
+        using namespace jtx;
+
+        const char credType[] = "abcde";
+        Account const issuer{"issuer"};
+        Account const alice{"alice"};
+        Account const bob{"bob"};
+        Account const john("john");
+        Account const maria{"maria"};
+
+        {
+            testcase("Payment with credentials.");
+
+            Env env(*this);
+
+            env.fund(XRP(5000), issuer, alice, bob);
+            env.close();
+
+            // Issuer create credentials, but Alice didn't accept them yet
+            env(credentials::create(alice, issuer, credType));
+            env.close();
+
+            // Get the index of the credentials
+            auto const jv =
+                credentials::ledgerEntry(env, alice, issuer, credType);
+            std::string const credIdx = jv[jss::result][jss::index].asString();
+
+            // Bob require preauthorization
+            env(fset(bob, asfDepositAuth));
+            env.close();
+
+            // Bob will accept payements from accounts with credentials signed
+            // by 'issuer'
+            env(deposit::authCredentials(bob, {{issuer, credType}}));
+            env.close();
+
+            auto const jDP =
+                ledgerEntryDepositPreauth(env, bob, {{issuer, credType}});
+            BEAST_EXPECT(
```

Comment:
The contents of the object should be checked here.

Reply:
added